syntax = "proto3";

import "gameactions.proto";
import "clientactions.proto";
import "serveractions.proto";
import "playeractions.proto";


package proto;

// deprecated message, here just to compile while we remove it completly from the client
message Action {
  string message = 1;
  int32 valeur = 2;
}

enum PGlobalActionType {
  CLIENT_ACTION = 0;
  PLAYER_ACTION = 1;// unsupported
  SERVER_ACTION = 2;
  GAME_ACTION = 3;
}

// The only protobuf message servers and clients can send, main container of any message
message PContainer {
  PGlobalActionType global_action = 1;
  PClientAction client_action = 2;
  PPlayerAction player_action = 3;
  PServerAction server_action = 4;
  PGameAction game_action = 5;
}